File: be-engine-v1/be-wallet/src/main/java/com/cmcorg20230301/be/engine/wallet/model/enums/SysUserWalletWithdrawStatusEnum.java
package com.cmcorg20230301.be.engine.wallet.model.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 用户钱包提现状态，枚举类
 */
@Getter
@AllArgsConstructor
public enum SysUserWalletWithdrawStatusEnum {

    COMMIT(101, "待受理"), // 待受理（可取消）

    ACCEPT(201, "受理中"), // 受理中（不可取消）

    SUCCESS(301, "已成功"), // 已成功

    REJECT(401, "已拒绝"), // 已拒绝（需要填写拒绝理由）

    CANCEL(501, "已取消"), // 已取消（用户在待受理的时候，可以取消）

    ;

    @EnumValue
    @JsonValue
    private final int code; // 类型编码

    private final String name;

}

File: fe-antd-v1/src/model/enum/SysUserWalletWithdrawStatusEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {PresetStatusColorType} from "antd/es/_util/colors";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";
import {SysUserWalletWithdrawLogDictListWithdrawStatus} from "@/api/http/SysUserWalletWithdrawLog";

export interface ISysUserWalletWithdrawStatusEnum {

    COMMIT: IEnum<number>,
    ACCEPT: IEnum<number>,
    SUCCESS: IEnum<number>,
    REJECT: IEnum<number>,
    CANCEL: IEnum<number>,

}

// 用户提现状态枚举类
export const SysUserWalletWithdrawStatusEnum: ISysUserWalletWithdrawStatusEnum = {

    COMMIT: {
        code: 101,
        name: '待受理', // 待受理（可取消）
        status: 'warning',
    },

    ACCEPT: {
        code: 201,
        name: '受理中', // 受理中（不可取消）
        status: 'processing',
    },

    SUCCESS: {
        code: 301,
        name: '已成功', // 已成功
        status: 'success',
    },

    REJECT: {
        code: 401,
        name: '已拒绝', // 已拒绝（需要填写拒绝理由）
        status: 'error',
    },

    CANCEL: {
        code: 501,
        name: '已取消', // 已取消（用户在待受理的时候，可以取消）
        status: 'default',
    },

}

export const SysUserWalletWithdrawStatusEnumMap = new Map<number, PresetStatusColorType>();

Object.keys(SysUserWalletWithdrawStatusEnum).forEach(key => {

    const item = SysUserWalletWithdrawStatusEnum[key] as IEnum<number>;

    SysUserWalletWithdrawStatusEnumMap.set(item.code!, item.status!)

})

// 设置：用户提现状态的字典
export function UpdateWithdrawStatusDict(setWithdrawStatusDict: (value: (((prevState: (Map<number, ProSchemaValueEnumType> | undefined)) => (Map<number, ProSchemaValueEnumType> | undefined)) | Map<number, ProSchemaValueEnumType> | undefined)) => void) {

    SysUserWalletWithdrawLogDictListWithdrawStatus().then(res => {

        const dictMap = new Map<number, ProSchemaValueEnumType>();

        res.data?.map((it) => {

            dictMap.set(it.id!, {text: it.name, status: SysUserWalletWithdrawStatusEnumMap.get(it.id!)})

        })

        setWithdrawStatusDict(dictMap)

    })

}
